apps/rush-lib/src/cli/actions/PublishAction.ts

```diff
@@ -478,8 +483,51 @@ export class PublishAction extends BaseRushAction {
     }
   }
 
+  private _addNpmPublishHome(): void {
+    // Example: "common\config\rush\.npmrc-publish"
+    const sourceNpmrcPublishPath: string = path.join(this.rushConfiguration.commonRushConfigFolder, '.npmrc-publish');
+
+    // Example: "common\temp\publish-home"
+    const targetNpmrcPublishFolder: string = path.join(this.rushConfiguration.commonTempFolder, 'publish-home');
+
+    // Example: "common\temp\publish-home\.npmrc"
+    const targetNpmrcPublishPath: string = path.join(targetNpmrcPublishFolder, '.npmrc');
+
+    try {
+      // Check if .npmrc-publish file exists to use for publishing
+      if (FileSystem.exists(sourceNpmrcPublishPath)) {
+        // Sync "common\config\rush\.npmrc-publish" --> "common\temp\publish-home\.npmrc"
+        Utilities.createFolderWithRetry(targetNpmrcPublishFolder);
+
+        // Copy down the committed .npmrc-publish file, if there is one
+        Utilities.copyAndTrimNpmrcFile(sourceNpmrcPublishPath, targetNpmrcPublishPath);
+      } else if (FileSystem.exists(targetNpmrcPublishPath)) {
+        // If the source .npmrc-publish doesn't exist and there is one in the target, delete the one in the target
+        console.log(`Deleting ${targetNpmrcPublishPath}`);
+        FileSystem.deleteFile(targetNpmrcPublishPath);
+      }
+    } catch (e) {
+      throw new Error(`Error syncing .npmrc-publish file: ${e}`);
+    }
```

Comment:
This code seems very similar to `Utilities.syncNpmrc`. Can we just add a parameter to that function for the `.npmrc` file name and deduplicate this logic (and maybe revert the refactor to that code to reduce churn)?

Reply:
> (and maybe revert the refactor to that code to reduce churn)?

Are you referring to the optional Boolean flag in `syncNpmrc()` I had in [`261e44f`](https://github.com/microsoft/rushstack/pull/1602/commits/261e44f74ce5992e2ab405e9c3f9f6b52238a03f)?

Reply:
I was referring to the change that extracted the a bunch of the functionality into a function.

Reply:
Just to be on the safe side, my previous commit, [`b25306c`](https://github.com/microsoft/rushstack/pull/1602/commits/b25306cdb4d6e638c6700799b02d162328095d70), that duplicated logic in `syncNpmrc()` logic?

Reply:
I was actually referring to this commit https://github.com/microsoft/rushstack/commit/edfe7aff025a6dd0db70486d58b3fec41d90dfb3, but in retrospect I'm cool with the refactor. Let's just deduplicate the sync code.

Reply:
Okay, sounds good.